io_scene_nif/modules/nif_import/collision/__init__.py

```diff
@@ -81,12 +80,18 @@ def set_b_collider(b_obj, bounds_type, radius, n_obj=None):
         override = bpy.context.copy()
         override['selected_objects'] = b_obj
         bpy.ops.rigidbody.object_add(override)
+        # alternate but not functional way without context override
```

Comment:
Yeah I was reading through the Blender API docs and they recommend to always copy the context.
See - https://docs.blender.org/api/current/bpy.ops.html#overriding-context

Reply:
Right, there doesn't seem to be a way of adding rigid bodies without using operators, so your way is the best for now.